packages/components/src/tooltip/test/index.js

```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
 	describe( '#render()', () => {
 		it( 'should not render the tooltip if multiple children are passed', () => {
 			render(
-				<Tooltip text="Help text">
-					<button>Button 1</button>
-					<button>Button 2</button>
+				<Tooltip { ...props }>
+					<Button>Button 1</Button>
+					<Button>Button 2</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByText( 'Button 1' );
-			act( () => button.focus() );
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
-		it( 'should render children', () => {
+		it( 'should not render the tooltip if there is no focus', () => {
 			render(
-				<Tooltip text="Help text">
-					<button>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
 			expect(
 				screen.getByRole( 'button', { name: 'Hover Me!' } )
 			).toBeVisible();
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
-		it( 'should render children with additional tooltip when focused', async () => {
-			const mockOnFocus = jest.fn();
+		it( 'should render the tooltip when focusing anchor via tab', async () => {
+			const user = userEvent.setup();
 
 			render(
-				<Tooltip text="Help text">
-					<button onFocus={ mockOnFocus }>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
+			expect( document.body ).toHaveFocus();
 
-			// Before focus, the tooltip is not shown.
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			await user.tab();
 
-			act( () => button.focus() );
+			expect(
+				screen.getByRole( 'button', { name: /Hover me!/i } )
+			).toHaveFocus();
 
-			// Tooltip is shown after focusing the anchor.
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
 			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
```

Comment:
This is being tested in a lot of tests. I still need to look into this further, to see why this is the case.

Reply:
This is because the tooltip takes some time to be displayed and put in the right position. This doesn't happen immediately, so we might want to verify that it has finished before continuing with our tests' assertions.

Reply:
In this case, though, there are no further assertions, and this is necessary in order to ensure that by the time we're finishing the test, there are no running animations or pending/scheduled updates. IIRC those pending updates are handled by an external library and we don't have much control over them, and they're set in a way that `@testing-library`'s post-test-cleanup doesn't properly handle them, causing bleeding between tests and subsequent test failure to occur.

Reply:
Thanks for sharing and for your initial comments on this draft! 🙇‍♀️

Reply:
From @tyxla's explanation above I assumed that `.toBeVisible()` wasn't enough to ensure the animations/updates have resolved, unless I'm misunderstanding?

Reply:
What you said, Brooke, that's my understanding as well.